### PR TITLE
ISPN-128806 BlockingManagerImpl#thenRunBlocking doesn't join

### PR DESCRIPTION
**Pull request by wburns**
https://issues.redhat.com/browse/ISPN-12806

### CODE REVIEW

File: core/src/main/java/org/infinispan/util/concurrent/BlockingManagerImpl.java [hunk 1]
```diff
@@ -197,6 +197,7 @@ public void execute(Runnable command) {
             log.tracef("Invoked thenRun on a blocking thread, joining %s in same blocking thread", traceId);
          }
          try {
+            CompletionStages.join(stage);
             return stage.thenRun(runnable);
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/util/concurrent/BlockingManagerImpl.java:201]: it isn't trivial for me xD
Why do you need `stage.thenRun(runnable)`? 
Could you just invoke `runnable.run()` and `return CompletableFutures.completedNull()`?

Comment (wburns) [reply core/src/main/java/org/infinispan/util/concurrent/BlockingManagerImpl.java:201]: Sure, either way would work. The point is the stage needs to be complete. I can change it to that if you want.

Comment (pruivo) [reply core/src/main/java/org/infinispan/util/concurrent/BlockingManagerImpl.java:201]: I would say to invoke the `Runnable` directly.
`java.util.concurrent.CompletableFuture#thenRun()` seems to create a new `CompletableFuture` instance.

Comment (wburns) [reply core/src/main/java/org/infinispan/util/concurrent/BlockingManagerImpl.java]: Yep, I already pushed the change. 👍

**Review by pruivo — APPROVED** [review]

**Comment by ryanemerson** [comment]
Thanks @wburns

**Pull request merged**